src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +31,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+
+        String[] addressDetails = address.split(",");        
+        this.block = new Block(addressDetails[0]);
+        this.street = new Street(addressDetails[1]);
+        this.unit = new Unit(addressDetails[2]);
+        this.postalCode = new PostalCode(addressDetails[3]);
```

Comment:
- how do you check for the equality of the address?
- how do you generate the hashcode for use later?